docs/buffering.md

```diff
@@ -0,0 +1,49 @@
+
+
+StreamInfo.streamCaps.maxLatency - specifies the duration in excess of which StreamLatencyPressureFunc callback will be called. The value should be anywhere between a couple of fragments duration to bufferDuration. If the value is 0 or greater than bufferDuration, no latency pressure callback will be issued. C Producer layer default logic will attempt to re-connect when this callback is fired potentially resulting in a better connection.
+
+
+DeviceInfo.storageInfo.storageSize - specifies the overall content store size to allocate for storing all of the buffered frames for all of the streams (in case the for a given Client object). The default behavior on overflowing the storage is to evict the tail frames until there is enough storage available to put a new frame. As the content store fills up and the utilization reaches 95%, content store pressure callback will be issued.
+
+
+### Dropping frames
+
+The SDK drops frames from the Content View in the following scenarios
+
+* If the stream is configured with persistence and the Persisted ACK is received.
+* When the frames naturally fall off the tail as the new frames are being added and the delta between the new frame and the oldest Tail frame is greater than the buffer_duration.
+
+In the first case, the ACK timestamp is mapped in the Content View timeline to find the Key-frame that the ACK applies to. The tail is then moved past to the next Key-frame to trim the workset as those frames are no longer needed (the fragment had already been durably persisted).
+
+In the latter case, the frames that get dropped at the tail might or might not yet have been sent out. The Tail frame could be partially sent out. The default policy controls on how the frames are being dropped. If the Tail frame has been sent out, it will be dropped without calling the dropped frame callback (success case). In case it's been partially sent, it will be preserved and attempted to be sent to ensure syntactic correctness of the resulting MKV stream. The following frames will be dropped until the next Key-frame to free up buffer space.
+
+The default policy applied to PutFrame API call on storage pressure is to attempt to evict the tail frames to make space for the newly produced frame.
+
+
+### Stream staleness
+
+KVS streaming abstracts the actual networking protocol and doesn't rely on the network-level ACKs. Moreover, the network level ACKs are still insufficient for guaranteeing the delivery of the video data to the backend. Stream staleness is calculated by measuring the time between the successive Buffering ACKs. If this time is above the threshold specified in StreamCaps (https://github.com/awslabs/amazon-kinesis-video-streams-pic/blob/master/src/client/include/com/amazonaws/kinesis/video/client/Include.h#L941) the StreamConnectionStaleFunc callback will be executed. The default handler for this callback will reset the connection but the applications could choose to perform other actions if needed.
```

Comment:
Is the default response to stream staleness similar to the response for max latency callback? If so, would be good to clarify that.

Reply:
Will do. And yes, it resets the connection

Reply:
Just saw that the last sentence is mentioning this